### ISSUE TITLE
[Bug] After countinueAsNew awaitWithTimeout doesn't check for condition

### ISSUE THREAD

**Issue by mixadior** [temporalio/sdk-php#122]
### Describe the bug

my workflow waits for some condition via yield Workflow::awaitWithTimeout , but after workflow "restart" via continueAsNewStub seems that condition never goes to true. Let me provide an example code. 

### To Reproduce

basic code used from https://github.com/wolfy-j/temporal-workshop

```php

<?php

namespace Workshop\App\Test;

use Temporal\Activity\ActivityOptions;
use Temporal\Workflow;
use Temporal\Workflow\WorkflowMethod;

class Test2Workflow implements Test2WorkflowInterface {

    private $photosBySourceEstateToProcess = [];
    private $photosProcessActivity;

    public function __construct()
    {
        $this->photosProcessActivity = Workflow::newActivityStub(
            TestActivity::class,
            ActivityOptions::new()->withStartToCloseTimeout(\DateInterval::createFromDateString('30 seconds'))
        );
    }


    public function run(array $initialPhotosToProcess = [])
    {
        if ($initialPhotosToProcess) {
            $this->photosBySourceEstateToProcess = $initialPhotosToProcess;
        }

        file_put_contents("php://stderr","startttt ".count($this->photosBySourceEstateToProcess));

        yield Workflow::awaitWithTimeout(300, fn()=> (count($this->photosBySourceEstateToProcess) > 10) || (count($initialPhotosToProcess) > 10));
        $photosToProcess = array_splice($this->photosBySourceEstateToProcess, 0, 10);

        foreach ($photosToProcess as $processData) {
            $processData = (array) $processData;
            yield $this->photosProcessActivity->loadSePhotos($processData['id'], $processData['photos'], $processData['withPage']);
        }

        return Workflow::newContinueAsNewStub(self::class)->run( $this->photosBySourceEstateToProcess);
    }
    #[Workflow\SignalMethod]
    public function setSourceEstatePhotos(int $sourceEstateId, $photos = [], $withPage = false)
    {
        $this->photosBySourceEstateToProcess[] = ['id' => $sourceEstateId, 'photos' => $photos, 'withPage' => $withPage];
    }

    #[Workflow\QueryMethod]
    public function getSourceEstatePhotos(): array
    {
        return $this->photosBySourceEstateToProcess;
    }
}

```

activity code

```php

<?php

namespace Workshop\App\Test;

class TestActivity implements TestActivityInterface {

    public function hello(string $name): string
    {
        return sprintf("Hello, %s", $name);
    }

    public function loadSePhotos(int $sourceEstateId, array $photos, bool $withPage): string
    {
        $rayId = uniqid("v");
        file_put_contents("php://stderr", sprintf("%s, Working on.Load se #%d photos. Photos: %s", $rayId, $sourceEstateId, join(",", $photos)));
        sleep(1);
        file_put_contents("php://stderr", sprintf("%s, done", $rayId));
        return "yes";
    }
}

```

command used to start workflow:

```php

<?php

namespace Workshop\App\Test;

use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Workshop\Util\Command;

class Test4Command extends Command {
    protected const NAME = 'run_source_photos';

    public function execute(InputInterface $input, OutputInterface $output)
    {
        try {
            $workflow = $this->workflowClient->newWorkflowStub(Test2Workflow::class);
            $d = $this->workflowClient->start($workflow);


            $output->writeln(sprintf("Id: %s", $d->getExecution()->getID()));
        } catch (\Throwable $ex) {
            $output->writeln($ex->getTraceAsString());
        }
        return 0;
    }
}

```

command used to send signals:

```php

<?php

namespace Workshop\App\Test;

use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Workshop\Util\Command;

class Test3Command extends Command {
    protected const NAME = 'push_photos';
    protected const ARGUMENTS = [
        ['name' => 'wfId']
    ];

    public function execute(InputInterface $input, OutputInterface $output)
    {
        try {
            $wf = $this->workflowClient->newRunningWorkflowStub(Test2Workflow::class, $input->getArgument('wfId'));

            for ($i = 0;$i < 30;$i++) {
                $wf->setSourceEstatePhotos($i, ["1.jpg", "2.jpg", "3.jpg"]);
            }

        } catch (\Throwable $ex) {
            $output->writeln($ex->getMessage());
            $output->writeln($ex->getTraceAsString());
        }
        return 0;
    }
}

```

so first i execute

```sh

php ./app.php run_source_photos

```

get an workflow id and run 

```sh

php ./app.php push_photos %workflow_id%

```

### Expected behavior

that right after processing of first 10 elements and workflow restart it will start next workflow and `yield Workflow::awaitWithTimeout` wouldn't wait for timeout, because condition will return true. 

### Screenshots/Terminal output

![](http://feodosian.com/public/mc_mixadiorMacBook-Pro-3.localVolumesWorkspaceprojectstemporal-workshop_2021-08-18_20-34-12.png)

### Versions

- OS: Linux (docker)
- Temporal Version: 1.9.0
- docker-compose - https://github.com/temporalio/docker-compose

### Additional context

.rr.yaml

```yaml

rpc:
  listen: tcp://127.0.0.1:6001

server:
  command: "php worker.php"
  env:
    XDEBUG_SESSION: 1
    XDEBUG_CONFIG: idekey=PHPSTORM
    PHP_IDE_CONFIG: serverName=temporalphp
    

logs:
  level: info
  channels:
    informer.mode: none

temporal:
  address: host.docker.internal:7233
  activities:
    debug: false
    num_workers: 8

http:
  address: 127.0.0.1:8080
  pool:
    num_workers: 8


```

**Comment by wolfy-j** [comment]
That you for the report. Added to the pipeline.

**Comment by wolfy-j** [comment]
This issue has been addressed in 1.0.5
